### PR TITLE
feat(BS-14503) Task list localization

### PR DESCRIPTION
**Pull request by cpuy**
Based on  #107

Add translate filter and directive, use gettextCatalog

Implement [BS-14503](https://bonitasoft.atlassian.net/browse/BS-14503)

I may have forgotten some strings (I am pretty I have) anyway this is a nice first shot (IMHO). I'll let associated Jira Open so when PR is merged, key translated in crowdin and integrated in build, I'll test this carefully

### CODE REVIEW

File: main/features/user/tasks/list/tasks-details.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <button class="pull-right btn btn-primary"
         id="Button{{editable? 'Release' : 'Take'}}DetailsColumn"
-        ng-click="onTakeReleaseTask()" title="{{editable ? 'Unassign the task and put it back in Available tasks' : 'Assign the task to myself and move it to My tasks.'}}">
+        ng-click="onTakeReleaseTask()" title="{{editable ? 'Unassign the task and put it back in Available tasks' : 'Assign the task to myself and move it to My tasks.' | translate}}">
   <i class="glyphicon" ng-class="{'glyphicon-save':!editable, 'glyphicon-open':editable}"></i>
-  <span class="sr-only">{{editable ? 'Release' : 'Take'}}</span>
+  <span class="sr-only">{{editable ? 'Release' : 'Take' | translate}}</span>
```
Comment (abirembaut) [inline main/features/user/tasks/list/tasks-details.html]: Does that work ? The translator might be asked to translate "editable", no ?

Comment (cpuy) [inline main/features/user/tasks/list/tasks-details.html]: I don't know :-)

IMHO what will be translated is `Release` or `Take` but I wonder what string will be extracted in pot file.

Let me see

**Comment by cpuy** [comment]
Keys are not extracted for things like that 
```
{{ condition ? 'key1' : 'key2' }}
```
I'll change that

File: main/features/user/tasks/list/tasks-details.html [hunk 2]
```diff
@@ -14,30 +21,30 @@ <h3 class="Context-title">{{currentTask.name}}</h3>
...
         {{currentTask.description}}
       </p>
-      <h4>Case Info</h4>
+      <h4 translate>Case Info</h4>
       <ul class="list-group">
         <li class="list-group-item" ng-repeat="supervisor in currentCase.supervisors">
-          <span class="ListGroup-label">Process Supervisor&nbsp;:</span>
-          <a ng-href="mailto:{{::supervisor.user_id.email}}" title="mail to {{::supervisor.user_id.name}}">
+          <span class="ListGroup-label" translate>Process Supervisor&nbsp;:</span>
+          <a ng-href="mailto:{{::supervisor.user_id.email}}" title="mailtoTitle(supervisor.user_id.name)">
```
Comment (cpuy) [inline main/features/user/tasks/list/tasks-details.html]: Wrong

File: main/features/user/tasks/list/tasks-list.html [hunk 3]
```diff
@@ -11,11 +11,11 @@
       'TaskListPage--small':app.smallScreen,
       'TaskListPage--withMenu':!app.smallScreen && app.showMenu}">
     <div class="OffcanvasMenu navbar">
-      <h3 class="OffcanvasMenu-title">Filters</h3>
-      <button ng-hide="app.smallScreen" title="{{app.showMenu ? 'Hide' : 'Show'}} Menu"
+      <h3 class="OffcanvasMenu-title" translate>Filters</h3>
+      <button ng-hide="app.smallScreen" title="{{app.showMenu ? 'Hide Menu' : 'Show Menu' | translate}}"
```
Comment (cpuy) [inline main/features/user/tasks/list/tasks-list.html]: Wrong

File: main/features/user/tasks/list/tasks-list.html [hunk 4]
```diff
@@ -111,11 +111,11 @@ <h3 class="OffcanvasMenu-title">Filters</h3>
                 <button class="pull-right btn btn-default"
                         ng-model="app.expandDetails"
                         btn-checkbox
-                        title="{{app.expandDetails ? 'Expand' : 'Reduce'}}">
+                        title="{{app.expandDetails ? 'Expand' : 'Reduce' | translate}}">
```
Comment (cpuy) [inline main/features/user/tasks/list/tasks-list.html]: Wrong

File: main/features/user/tasks/list/tasks-list.html [hunk 5]
```diff
@@ -111,11 +111,11 @@ <h3 class="OffcanvasMenu-title">Filters</h3>
...
                   <span ng-class="{'glyphicon glyphicon-resize-full':!app.expandDetails,'glyphicon glyphicon-resize-small':app.expandDetails}"></span>
-                  <span class="sr-only">{{app.expandDetails ? 'Expand' : 'Reduce'}}</span>
+                  <span class="sr-only">{{app.expandDetails ? 'Expand' : 'Reduce' | translate}}</span>
```
Comment (cpuy) [inline main/features/user/tasks/list/tasks-list.html]: Wrong

**Pull request merged**